packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskScheduler.kt

```diff
@@ -0,0 +1,214 @@
+      val builder = PeriodicWorkRequestBuilder<BackgroundTaskWork>(
+        repeatIntervalTimeUnit = TimeUnit.MINUTES,
+        repeatInterval = intervalMinutes
+      ).setInputData(data.build())
+        .setConstraints(
+          Constraints.Builder()
+            .setRequiresBatteryNotLow(true)
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+        )
+
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        // Add minimum interval here as well so that the work doesn't start immediately
+        // In Debug builds we wait 30 seconds to make sure background tasks are easy to test.
+        // Remember to put the app in the background for the task to run.
+        if (ReactBuildConfig.DEBUG) {
+          builder.setInitialDelay(Duration.ofSeconds(30))
+        } else {
+          builder.setInitialDelay(Duration.ofMinutes(intervalMinutes))
+        }
+      }
+
+      // Create work request
+      val workRequest = builder.build()
+
+      // Get Work manager
+      val workManager = WorkManager.getInstance(context)
+
+      // Enqueue the work
+      return try {
+        val operation = workManager.enqueueUniquePeriodicWork(
+          WORKER_IDENTIFIER,
+          ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE,
+          workRequest
+        ).await()
+
+        Log.i(TAG, "Worker enqueued successfully")
+        true
+      } catch (e: Exception) {
+        Log.e(TAG, "Worker failed to start with error " + e.message)
+        false
+      }
+    }
+
+    /**
+     Cancels the worker task
+     */
+    suspend fun stopWorker(context: Context): Boolean {
+      Log.i(TAG, "Cancelling worker with identifier $WORKER_IDENTIFIER")
+
+      // Stop our main worker
+      val workManager = WorkManager.getInstance(context)
+      return try {
+        workManager.cancelUniqueWork(WORKER_IDENTIFIER).await()
+        workManager.pruneWork().await()
+      } catch (e: Exception) {
+        Log.i(TAG, "Stopping worker failed with error " + e.message)
+        false
+      }
+    }
+
+    /**
+     Returns true if the worker task is pending
+     */
+    suspend fun isWorkerRunning(context: Context): Boolean {
+      val workInfo = getWorkerInfo(context)
+      return workInfo?.state == WorkInfo.State.RUNNING ||
+        workInfo?.state == WorkInfo.State.ENQUEUED
+    }
+
+    /**
+     * Runs tasks with the given appScopeKey
+     */
+    suspend fun runTasks(context: Context, appScopeKey: String) {
+      // Get task service
+      val taskService = TaskServiceProviderHelper.getTaskServiceImpl(context)
+        ?: throw MissingTaskServiceException()
+
+      Log.i(TAG, "runTasks: $appScopeKey")
+
+      // Get all task consumers
+      val consumers = taskService.getTaskConsumers(appScopeKey)
+      Log.i(TAG, "runTasks: number of consumers ${consumers.size}")
+
+      val tasks = consumers.mapNotNull { consumer ->
+        if (consumer.taskType() == BackgroundTaskConsumer.BACKGROUND_TASK_TYPE) {
```

Comment:
Instead of checking `taskType`, can't we just try to cast it to `BackgroundTaskConsumer`?